packages/material-ui/src/Paper/Paper.js

```diff
@@ -56,6 +56,16 @@ const Paper = React.forwardRef(function Paper(props, ref) {
     'MuiPaper',
   );
 
+  if (process.env.NODE_ENV !== 'production') {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const theme = useTheme();
+    if (theme.shadows[elevation] === undefined) {
+      console.warn(
+        `Material-UI: The elevation provided <Paper elevation={${elevation}}> is not available in the theme.`,
```

Comment:
Maybe?

```suggestion
        [
          `Material-UI: The elevation provided <Paper elevation={${elevation}}> is not available in the theme.`,
          `Please make sure that \`theme.shadows[${elevation}]\` is defined.`,
        ]
```

Reply:
Thank you for suggesting better message!